dev-server/documents/html/whitespace.mustache

```diff
@@ -0,0 +1,107 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <title>Document with whitespace examples</title>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <style type="text/css">
+      p {
+        border: 1px dashed grey;
+        padding: 1em;
+      }
+
+      span {
+        border: 1px dotted pink;
+        padding: 0.5em;
+      }
+    </style>
+  </head>
+  <body>
+  <h1>HTML document with examples of whitespace</h1>
+  <p>This HTML document contains various types of whitespace, including HTML entities
+  representing Unicode whitespace characters, as well as empty elements.
+  It is intended to help test text selection.</p>
+    <div id="toplevel">
+      <div id="apple">
+        <p></p>
+        <p>
+          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam
+          fringilla ultrices lacus consequat pretium.
+          <span>Curabitur posuere <i>posuere</i> <span></span></span>
+          rhoncus. <b>Maecenas</b> sit <span></span>amet velit<span
+          > </span> eget arcu aliquam pretium sed a tellus. Praesent sodales
+          enim id justo bibendum condimentum sit amet sit amet lorem.
+        </p>
+
+        <p>&nbsp;h<span> &#8195;i &#8195;</span>i</p>
```

Comment:
It looks like the various paragraphs are trying to exercise slightly different cases. If so can you add a note above each. This will be useful in case we need to reference one in a bug report.

Reply:
I wasn't quite that rigorous (that is, there isn't necessarily a specific test case or scenario implied by each paragraph); I was looking for a collection of different combinations of whitespace characters, Unicode/entity-wise, and also wanted to create some visible (ergo the CSS) but "empty" inline and block elements. I'll see if I can add a few clarifying notes, sure.